core/tests/test_utils.py

```diff
@@ -1838,6 +1843,123 @@ def swap(self, obj, attr, newvalue):
         finally:
             setattr(obj, attr, original)
 
+    @contextlib.contextmanager
+    def swap_with_checks(
+            self, obj, attr, newvalue, expected_args=None, expected_kwargs=None,
+            called=True, called_times=None,
+            expected_exception=IntendedException):
+        """Swap an object's function value within the context of a
```

Comment:
This docstring is a bit messy. Format it as: general description, examples, note -- and then have Args and Yields sections, in order, at the end. Separate each pair of adjacent sections with a single newline.

Reply:
Done